packages/capacitor-plugin/android/src/main/java/com/capacitorjs/plugins/filetransfer/FileTransferPlugin.kt

```diff
@@ -0,0 +1,190 @@
+            map[key] = getString(key) ?: ""
+        }
+        return map
+    }
+
+    private fun JSObject.toParamsMap(): Map<String, Array<String>> {
+        val map = mutableMapOf<String, Array<String>>()
+        keys().forEach { key ->
+            val value = getString(key)
+            if (value != null) {
+                map[key] = arrayOf(value)
+            }
+        }
+        return map
+    }
+
+    @PluginMethod
+    fun downloadFile(call: PluginCall) {
+        val url = call.getString("url") ?: run {
+            call.reject("URL is required")
+            return
+        }
+        val relativePath = call.getString("path") ?: run {
+            call.reject("Path is required")
+            return
+        }
+        val progress = call.getBoolean("progress", false) ?: false
+
+        val headers = call.getObject("headers") ?: JSObject()
+        val params = call.getObject("params") ?: JSObject()
+
+        val httpOptions = IONFLTRTransferHttpOptions(
+            method = call.getString("method") ?: "GET",
+            headers = headers.toMap(),
+            params = params.toParamsMap(),
+            shouldEncodeUrlParams = call.getBoolean("shouldEncodeUrlParams", true) ?: true,
+            readTimeout = call.getInt("readTimeout", 60000) ?: 60000,
+            connectTimeout = call.getInt("connectTimeout", 60000) ?: 60000,
+            disableRedirects = call.getBoolean("disableRedirects", false) ?: false
+        )
+
+        val options = IONFLTRDownloadOptions(
+            url = url,
+            filePath = relativePath,
+            httpOptions = httpOptions
+        )
+
+        ioScope.launch {
+            controller.downloadFile(options)
+                .onEach { result ->
+                    when (result) {
+                        is IONFLTRTransferResult.Ongoing -> {
+                            if (progress) {
+                                val progressData = JSObject().apply {
+                                    put("type", "download")
+                                    put("url", url)
+                                    put("bytes", result.status.bytes)
+                                    put("contentLength", result.status.contentLength)
+                                    put("lengthComputable", result.status.lengthComputable)
+                                }
+                                notifyListeners("progress", progressData)
+                            }
+                        }
+                        is IONFLTRTransferResult.Complete -> {
+                            val response = JSObject().apply {
+                                put("path", relativePath)
+                            }
+                            call.resolve(response)
+                        }
+                    }
+                }
+                .catch { error ->
+                    call.reject(error.message ?: "Unknown error occurred")
+                }
+                .collect{}
+        }
```

Comment:
Minor: Instead of launching a coroutine a doing `collect{}`, you can do

```suggestion
                .launchIn(ioScope)
        }
```